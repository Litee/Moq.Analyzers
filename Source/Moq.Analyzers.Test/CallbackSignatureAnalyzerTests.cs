using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.CodeFixes;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class CallbackSignatureAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldPassIfGoodParameters()
        {
            VerifyCSharpDiagnostic(File.ReadAllText("Data/CallbackSignatures/CallbackWithGoodParameters.cs"));
        }

        [Fact]
        public void ShouldFailIfBadParameters()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/CallbackSignatures/CallbackWithBadParameters.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new CallbackSignatureAnalyzer();
        }
    }
}