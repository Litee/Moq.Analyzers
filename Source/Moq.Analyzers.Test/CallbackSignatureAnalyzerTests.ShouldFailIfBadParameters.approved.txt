Diagnostic 1
	Id: MOQ1001
	Location: SourceFile(Test0.cs[432..439))
	Code: (int i)
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

Diagnostic 2
	Id: MOQ1001
	Location: SourceFile(Test0.cs[513..535))
	Code: (string s1, string s2)
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

Diagnostic 3
	Id: MOQ1001
	Location: SourceFile(Test0.cs[648..667))
	Code: (string s1, int i1)
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

Diagnostic 4
	Id: MOQ1001
	Location: SourceFile(Test0.cs[747..754))
	Code: (int i)
	Severity: Warning
	Message: Callback must have the same signature as the mocked method.

